### PR TITLE
Icon update

### PR DESCRIPTION
**Pull request by rasendubi**
See #57 and commit messages for the full story.

### CODE REVIEW

File: app/modules/tabsTracker.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import Immutable from 'immutable';
```
Comment (drets) [inline app/modules/tabsTracker.js:1]: I am not sure if `modules` folder is the best place for this file.

File: app/modules/tabsTracker.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    if (changeInfo.url) {
+      action.url = changeInfo.url; // eslint-disable-line
+    }
+    if (changeInfo.title) {
+      action.title = changeInfo.title; // eslint-disable-line
+    }
+    if (changeInfo.status) {
+      action.status = changeInfo.status; // eslint-disable-line
+    }
+    store.dispatch(action);
```
Comment (drets) [inline app/modules/tabsTracker.js:34]: I'd write it in more fancy way:
```js
    store.dispatch({
      type: 'TAB_CHANGED',
      tabId,
      ...changeInfo.url && { url: changeInfo.url },
      ...changeInfo.title && { title: changeInfo.title },
      ...changeInfo.status && { status: changeInfo.status },
    });
```

File: app/modules/tabsTracker.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  activeTabId: null,
+});
+
+export const getCurrentTab = (store) => store.getIn(['tabs', 'activeTabId']);
+
+export const getTabUrl = (store, tabId) => store.getIn(['tabs', 'tabs', tabId, 'url']);
+
+export const getTabs = (store) => store.getIn(['tabs', 'tabs']);
+
+export const getTab = (store, tabId) => store.getIn(['tabs', 'tabs', tabId], new Immutable.Map());
```
Comment (drets) [inline app/modules/tabsTracker.js:14]: Why tabs-tabs?

File: app/modules/tabsTracker.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  chrome.tabs.onRemoved.addListener((tabId, removeInfo) => { // eslint-disable-line
+    store.dispatch({
+      type: 'TAB_REMOVED',
+      tabId,
+    });
+  });
+};
+
+const updateState = (state, action) =>
+      state.mergeIn(['tabs', action.tabId], Immutable.fromJS(action).delete('type').delete('tabId'));
```
Comment (drets) [inline app/modules/tabsTracker.js:58]: Why delete?

File: app/selectors/view.js [hunk 5]
```diff
@@ -1,3 +1,3 @@
 import Immutable from 'immutable';
 
-export const getViewState = (state, viewId) => state.getIn(['dataView', viewId]) || new Immutable.Map();
+export const getViewState = (state, viewId) => Immutable.fromJS(state).getIn(['dataView', viewId]) || new Immutable.Map();
```
Comment (drets) [inline app/selectors/view.js:3]: [Here](https://github.com/rasendubi/heutagogy-chrome-extension/blob/a64b4119683b3b47d593568fa716d1833ed16c6e/app/components/LoginForm/LoginForm.js#L133) we're already passing immutable state, why do we need to call `fromJS` second time?

**Review by drets — COMMENTED** [review]
Good job, @rasendubi :+1:

File: app/selectors/view.js (hunk 5 above)
Comment (rasendubi) [reply app/selectors/view.js:3]: Is it `Immutable` in all usages?

File: app/modules/tabsTracker.js (hunk 4 above)
Comment (rasendubi) [reply app/modules/tabsTracker.js:58]: So we don't store type and tabId along with a bookmark. (So they don't cause spurious icon updates.)

File: app/modules/tabsTracker.js (hunk 2 above)
Comment (rasendubi) [reply app/modules/tabsTracker.js:34]: Nice. I don't know js that well)

File: app/selectors/view.js (hunk 5 above)
Comment (drets) [reply app/selectors/view.js:3]: I suppose − no, but then the question why, why we can't have it the same in all the cases :thinking:

File: app/modules/tabsTracker.js (hunk 3 above)
Comment (rasendubi) [reply app/modules/tabsTracker.js:14]: all `tabTracker` data is stored inside top-level `tab` field

File: app/modules/tabsTracker.js (hunk 1 above)
Comment (rasendubi) [reply app/modules/tabsTracker.js:1]: I'm just experimenting with alternate structure :)

File: app/modules/tabsTracker.js (hunk 3 above)
Comment (drets) [reply app/modules/tabsTracker.js:14]: Maybe top one rename to tabTracker?

**Pull request merged**